verilog/tools/kythe/kythe_facts_extractor.cc

```diff
@@ -257,6 +289,34 @@ void KytheFactsExtractor::Visit(const IndexingFactNode& node) {
   }
 }
 
+void KytheFactsExtractor::ExtractFileList(const IndexingFactNode& file_list) {
+  std::vector<ScopeResolver> scope_resolvers;
+
+  for (const IndexingFactNode& root : file_list.Children()) {
+    // Create a new ScopeResolver and give the ownership to the scope_resolvers
+    // vector so that it can outlive KytheFactsExtractor.
+    // The ScopeResolver-s are created and linked together as a linked-list
+    // structure so that the current ScopeResolver can search for definitions in
+    // the previous files' scopes.
+    std::string file_path = GetFilePathFromRoot(root);
+    if (scope_resolvers.empty()) {
+      scope_resolvers.push_back(ScopeResolver(nullptr));
+    } else {
```

Comment:
This is only true in the first iteration, could this be added outside of the for-loop?